_posts/work/2019-09-02-interactive-hpc.md

```diff
@@ -0,0 +1,170 @@
+   already have in place, including two-factor security tokens.
+
+   This gives users the visual rich UI/UX that they've come to expect,
+   while still giving system administrators the security and control that they
+   need in order to maintain stability over the system.
+
+   Several large and well respected supercomputing centers have deployed
+   JupyterHub internally, including NERSC, SDSC, NCAR, and many others.
+
+For more information, interested readers may want to watch this talk from
+Anderson Banirhiwe at SciPy 2019 on using JupyterHub from within their
+supercomputing center.
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/vhawO8fgD64" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+
+
+### Elastic scaling and On-demand computing
+
+Data scientist's workloads are both bursty, and intolerant to delays.
+
+Often they want to quickly spin up 50 machines to churn through 100 TB of data
+for a five minutes, generate a plot, and then stare at that plot for an hour.
+Then, when they get an idea, they want to do that again, right now.
+They're comfortable waiting for a few mninutes, but if there is an
+hour-long wait in the queue then they're going to switch off to something
+else.
+
+These bursty workloads don't fit into most HPC job scheduling policies.
+These policies are designed to optimnize not for on-demand computing, but for
+high utilization and batch jobs.  This policy causes some users a great deal of
+pain because it means long wait times even for very short jobs.
+
+Today there are a few solutions to this, but they aren't great
+
+-  **Users** end up doing one of the following:
+
+    1.  Download a sample of the dataset locally and abandon their HPC resources
+    2.  Keep their 50 machines running the entire time, wasting valuable resources
+    3.  Get frustrated, and buy some AWS credits
+
+    That is, unless of course there are some gaps in the schedule, and if IT
+    have taken some steps to promote small jobs.
+
+    Most HPC machines don't hit 100% utilization.  There are little small gaps
+    in the schedule which, if your jobs are fast, you can sneak into.  A good
+    analogy I've heard before is that if you fill a bucket full of rocks, you
+    can still pour in a fair amount of sand.  Our goal is to be like sand.
+
+    Distributed systems like [Dask](https://dask.org),
+    are designed to work with job-queue schedulers (PBS/SLURM/SGE/LSF/...)
+    to submit many small fast jobs and assemble those jobs into a broader
+    distributed computing network.
+
+-  **System administrators** can improve the situation here by making a few
+   small policy decisions:
+
+    1.  Promote very short jobs to the front of the queue.  This is a common option in many HPC job queuing systems.
+    2.  Consider creating and advertising high priority queues that are designed
```

Comment:
:+1:

This would be a huge deal, especially in conjunction with `cluster.adapt()`